typed/reconcile_schema.go

```diff
@@ -283,6 +276,13 @@ func fieldSetAtPath(node *fieldpath.Set, path fieldpath.Path) (*fieldpath.Set, b
 	return node, ok
 }
 
+func descendToPath(node *fieldpath.Set, path fieldpath.Path) *fieldpath.Set {
```

Comment:
Nit for next time we look at it, this code should probably live in `fieldpath` rather than here.